packages/ai/core/middleware/extract-reasoning-middleware.ts

```diff
@@ -12,22 +12,26 @@ import { LanguageModelV1Middleware } from './language-model-v1-middleware';
 export function extractReasoningMiddleware({
   tagName,
   separator = '\n',
+  startWithReasoning = false,
 }: {
   tagName: string;
   separator?: string;
+  startWithReasoning?: boolean;
 }): LanguageModelV1Middleware {
   const openingTag = `<${tagName}>`;
   const closingTag = `<\/${tagName}>`;
 
   return {
     middlewareVersion: 'v1',
     wrapGenerate: async ({ doGenerate }) => {
-      const { text, ...rest } = await doGenerate();
+      const { text: rawText, ...rest } = await doGenerate();
 
-      if (text == null) {
-        return { text, ...rest };
+      if (rawText == null) {
+        return { text: rawText, ...rest };
       }
```

Comment:
I wonder why this is even there... text should always be non-nullish. hm

Reply:
Discussed here: https://vercel.slack.com/archives/C050WU03V3N/p1739977141153109?thread_ts=1739915413.936349&cid=C050WU03V3N